nexus-repository-composer-it/src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerProxyIT.java

```diff
@@ -81,16 +120,58 @@ public void unresponsiveRemoteProduces404() throws Exception {
     assertThat(status(proxyClient.get(BAD_PATH)), is(HttpStatus.NOT_FOUND));
   }
 
+  public void retrievePackagesJSONFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(FILE_PACKAGES)), is(HttpStatus.OK));
+
+    final Asset asset = findAsset(proxyRepo, FILE_PACKAGES);
+    assertThat(asset.name(), is(equalTo(FILE_PACKAGES)));
+    assertThat(asset.contentType(), is(equalTo(MIME_TYPE_JSON)));
+    assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
+  }
+
   @Test
-  public void retrieveProjectJSONFromProxyWhenRemoteOnline() throws Exception {
-    assertThat(status(proxyClient.get(VALID_PACKAGE_URL)), is(HttpStatus.OK));
+  public void providersURLChangedToNXRM() throws Exception {
+    assertThat(status(proxyClient.get(FILE_PACKAGES)), is(HttpStatus.OK));
+
+    try (CloseableHttpResponse response = proxyClient.get(FILE_PACKAGES)) {
+      HttpEntity entity = response.getEntity();
+      JsonElement element = new JsonParser().parse(IOUtils.toString(entity.getContent()));
+      JsonObject json = element.getAsJsonObject();
+
+      assertThat(json.get("providers-url").toString(), is(equalTo("\"http://localhost:10000/repository/composer-test-proxy/p/%package%.json\"")));
+    }
+  }
+
+  public void retrieveListJSONFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(VALID_LIST_URL)), is(HttpStatus.OK));
 
-    final Asset asset = findAsset(proxyRepo, VALID_PACKAGE_URL);
-    assertThat(asset.name(), is(equalTo(VALID_PACKAGE_URL)));
+    final Asset asset = findAsset(proxyRepo, FILE_LIST);
+    assertThat(asset.name(), is(equalTo(FILE_LIST)));
     assertThat(asset.contentType(), is(equalTo(MIME_TYPE_JSON)));
     assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
   }
 
+  @Test
+  public void retrieveProviderJSONFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(VALID_PROVIDER_URL)), is(HttpStatus.OK));
+
+    final Asset asset = findAsset(proxyRepo, VALID_PROVIDER_URL);
+    assertThat(asset.name(), is(equalTo(VALID_PROVIDER_URL)));
+    assertThat(asset.contentType(), is(equalTo(MIME_TYPE_JSON)));
+    assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
+  }
+
+  // TODO: Dude, this test, what the heck! It's completely wacky, someone needs to look at this
```

Comment:
I'm not sure, but wondering if [this code](https://github.com/sonatype-nexus-community/nexus-repository-composer/blob/moreit/nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerProxyFacetImpl.java#L184) that adds a `p/` prefix to the zip-ball url is part of the issue with this wacky test? After the call to [buildProviderPath()](https://github.com/sonatype-nexus-community/nexus-repository-composer/blob/moreit/nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerPathUtils.java#L100), the [PROVIDER_JSON_PATH](https://github.com/sonatype-nexus-community/nexus-repository-composer/blob/moreit/nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerPathUtils.java#L35) formatting makes the URL request.path look like: 
 ```
/p/rjkip/ftp-php.json
```

Not at all sure if that is intended or not.

Reply:
Ah! Keen eye! So I had to refresh my memory but it looks like we are handling the `/p/` for the provider correctly. The wacky test is for the zip file (we named it zipball because of matcher method names done previously by others) looks like the zip path is somehow broken -- still need to do a bit more exploring.